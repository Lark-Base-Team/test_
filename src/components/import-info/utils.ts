import { ref } from "vue";
import { i18n } from "@/i18n";
import { lifeCircleEventParams } from "@/utils/import/lifeCircle";

export interface stage {
  state: string;
  disabled: boolean;
  message?: string;
  progress: boolean;
  stage: string;
  success: number;
  error: number;
  number: number;
  name: string | number;
  title: string;
  index: number;
}

const stageIndex = {
  checkFieldTypes: 0,
  checkOptions: 1,
  setOptions: 2,
  analysisRecords: 3,
  updateRecords: 4,
  deleteRecords: 5,
  addRecords: 6,
};

export const currentStage = ref(0);

export const defaultStages = () => [
  {
    state: "waiting",
    disabled: false,
    message: "",
    progress: true,
    stage: "checkFieldTypes",
    success: 0,
    error: 0,
    number: 0,
    name: stageIndex["checkFieldTypes"],
    title: "importInfo.checkFieldTypes",
    index: stageIndex["checkFieldTypes"],
  },
  {
    state: "waiting",
    disabled: true,
    message: "",
    progress: false,
    stage: "checkOptions",
    success: 0,
    error: 0,
    number: 0,
    name: stageIndex["checkOptions"],
    title: "importInfo.checkOptions",
    index: stageIndex["checkOptions"],
  },
  // {
  //   state: "waiting",
  //   disabled: true,
  //   message: "",
  //   progress: false,
  //   stage: "setOptions",
  //   success: 0,
  //   error: 0,
  //   number: 0,
  //   name: stageIndex["setOptions"],
  //   title: "importInfo.setOptions",
  //   index: stageIndex["setOptions"],
  // },
  {
    state: "waiting",
    disabled: true,
    message: "",
    progress: false,
    stage: "analysisRecords",
    success: 0,
    error: 0,
    number: 0,
    name: stageIndex["analysisRecords"],
    title: "importInfo.analysisRecords",
    index: stageIndex["analysisRecords"],
  },
  {
    state: "waiting",
    disabled: true,
    message: "",
    progress: false,
    stage: "updateRecords",
    success: 0,
    error: 0,
    number: 0,
    name: stageIndex["updateRecords"],
    title: "importInfo.updateRecords",
    index: stageIndex["updateRecords"],
  },
  {
    state: "waiting",
    disabled: true,
    message: "",
    progress: false,
    stage: "deleteRecords",
    success: 0,
    error: 0,
    number: 0,
    name: stageIndex["deleteRecords"],
    title: "importInfo.deleteRecords",
    index: stageIndex["deleteRecords"],
  },
  {
    state: "waiting",
    disabled: true,
    message: "",
    progress: false,
    stage: "addRecords",
    success: 0,
    error: 0,
    number: 0,
    name: stageIndex["addRecords"],
    title: "importInfo.addRecords",
    index: stageIndex["addRecords"],
  },
];

export const stages = ref<stage[]>(defaultStages());

export function beforeCheckFields(e: lifeCircleEventParams) {
  const { number } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["checkFieldTypes"]
  );
  stages.value[index].number = number;
  if (number > 0) {
    stages.value[index].state = "loading";
    stages.value[index].progress = true;
  } else {
    stages.value[index].state = "success";
    stages.value[index].progress = false;
  }

  stages.value[index].disabled = false;
  currentStage.value = stageIndex["checkFieldTypes"];
}

const optionsFieldsNumber = ref(0);

export function onCheckFields(e: lifeCircleEventParams) {
  const { res } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["checkFieldTypes"]
  );
  stages.value[index].success += 1;
  if (res) {
    optionsFieldsNumber.value += 1;
    stages.value[index].message = i18n.global.t(
      "importInfo.checkFieldsMessage",
      {
        totalNumber: optionsFieldsNumber.value,
      }
    );
  }
  if (
    stages.value[index].success + stages.value[index].error ===
    stages.value[index].number
  ) {
    stages.value[index].progress = false;
    stages.value[index].state = "success";
  }
}

export function beforeCheckOptions(e: lifeCircleEventParams) {
  const { number } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["checkOptions"]
  );
  stages.value[index].number = number;
  if (number > 0) {
    stages.value[index].state = "loading";
    stages.value[index].progress = true;
  } else {
    stages.value[index].state = "success";
    stages.value[index].progress = false;
  }
  stages.value[index].disabled = false;
  currentStage.value = stageIndex["checkOptions"];
}

export function onCheckOptions(e: lifeCircleEventParams) {
  const { selects } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["checkOptions"]
  );
  console.log("onCheckOptions", selects, stages.value[index]);
  stages.value[index].success += 1;
  stages.value[index].message = i18n.global.t(
    "importInfo.checkOptionsMessage",
    {
      totalNumber: selects.length,
    }
  );
  if (
    stages.value[index].success + stages.value[index].error ===
    stages.value[index].number
  ) {
    stages.value[index].progress = false;
    stages.value[index].state = "success";
  }
}

// export function beforeSetOptions(e: lifeCircleEventParams) {
//   const { number } = e.data;
//   const index = stages.value.findIndex(
//     (stage) => stage.index === stageIndex["setOptions"]
//   );
//   stages.value[index].number = number;
//   if (number > 0) {
//     stages.value[index].state = "loading";
//     stages.value[index].progress = true;
//   } else {
//     stages.value[index].state = "success";
//   }
//   stages.value[index].disabled = false;
//   currentStage.value = stageIndex["setOptions"];
// }

// export function onSetOptions(e: lifeCircleEventParams) {
//   const { record } = e.data;
//   const index = stages.value.findIndex(
//     (stage) => stage.index === stageIndex["setOptions"]
//   );
//   if (record) {
//     stages.value[index].success += 1;
//   } else {
//     stages.value[index].error += 1;
//   }
// }

export function beforeAnalysisRecords(e: lifeCircleEventParams) {
  const { number } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["analysisRecords"]
  );
  stages.value[index].number = number;
  if (number > 0) {
    stages.value[index].state = "loading";
    stages.value[index].progress = true;
  } else {
    stages.value[index].state = "success";
    stages.value[index].progress = false;
  }
  stages.value[index].disabled = false;
  currentStage.value = stageIndex["analysisRecords"];
}

export function onAnalysisRecords(e: lifeCircleEventParams) {
  const { updateNumber, addNumber, message } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["analysisRecords"]
  );
  stages.value[index].success = updateNumber + addNumber;
  stages.value[index].message = message;
  if (stages.value[index].success === stages.value[index].number) {
    stages.value[index].progress = false;
    stages.value[index].state = "success";
  }
}

export function beforeUpdateRecords(e: lifeCircleEventParams) {
  const { updateList } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["updateRecords"]
  );
  stages.value[index].number = updateList.length;
  if (updateList.length > 0) {
    stages.value[index].state = "loading";
    stages.value[index].progress = true;
  } else {
    stages.value[index].state = "success";
    stages.value[index].progress = false;
  }
  stages.value[index].disabled = false;
  currentStage.value = stageIndex["updateRecords"];
}

export function onUpdateRecords(e: lifeCircleEventParams) {
  const { res } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["updateRecords"]
  );
  if (res) {
    stages.value[index].success += 500;
  } else {
    stages.value[index].error += 500;
  }
  if (
    stages.value[index].success + stages.value[index].error ===
    stages.value[index].number
  ) {
    stages.value[index].progress = false;
    stages.value[index].state = "success";
  }
}

export function beforeDeleteRecords(e: lifeCircleEventParams) {
  const { deleteList } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["deleteRecords"]
  );
  stages.value[index].number = deleteList.length;
  if (deleteList.length > 0) {
    stages.value[index].state = "loading";
    stages.value[index].progress = true;
  } else {
    stages.value[index].state = "success";
    stages.value[index].progress = false;
  }
  stages.value[index].disabled = false;
  currentStage.value = stageIndex["deleteRecords"];
}

export function onDeleteRecords(e: lifeCircleEventParams) {
  const { res } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["deleteRecords"]
  );
  if (res) {
    stages.value[index].success += 500;
  } else {
    stages.value[index].error += 500;
  }
  if (
    stages.value[index].success + stages.value[index].error ===
    stages.value[index].number
  ) {
    stages.value[index].progress = false;
    stages.value[index].state = "success";
  }
}

export function beforeAddRecords(e: lifeCircleEventParams) {
  const { number } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["addRecords"]
  );
  stages.value[index].number = number;
  stages.value[index].message = i18n.global.t("importInfo.addRecordsMessage");
  if (number > 0) {
    stages.value[index].state = "loading";
    stages.value[index].progress = true;
  } else {
    stages.value[index].state = "success";
    stages.value[index].progress = false;
  }
  stages.value[index].disabled = false;
  currentStage.value = stageIndex["addRecords"];
}

export function onAddRecords(e: lifeCircleEventParams) {
  const { res } = e.data;
  const index = stages.value.findIndex(
    (stage) => stage.index === stageIndex["addRecords"]
  );
  if (res) {
    stages.value[index].success += 500;
  } else {
    stages.value[index].error += 500;
  }
  if (
    stages.value[index].success + stages.value[index].error ===
    stages.value[index].number
  ) {
    stages.value[index].state = "success";
  }
}
