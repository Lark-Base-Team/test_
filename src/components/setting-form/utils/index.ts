import { FieldType } from "@lark-base-open/js-sdk";

export const ignoreFieldType = [
  FieldType.Lookup,
  FieldType.CreatedTime,
  FieldType.ModifiedTime,
  FieldType.Formula,
  FieldType.DuplexLink,
  FieldType.SingleLink,
  FieldType.CreatedUser,
  FieldType.ModifiedUser,
  FieldType.NotSupport,
  FieldType.Location,
  FieldType.AutoNumber,
  FieldType.GroupChat,
  // FieldType.User,
  FieldType.Denied,
  FieldType.Attachment,
];
