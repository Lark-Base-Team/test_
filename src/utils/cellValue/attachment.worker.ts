async function downloadFile(url: string) {
  return new Promise<File>((resolve, reject) => {
    const xhr = new XMLHttpRequest()
    xhr.open("GET", url, true)
    xhr.responseType = "blob"
    xhr.onload = function () {
      if (this.status === 200) {
        resolve(this.response)
      } else {
        reject(this.response)
      }
    }
    xhr.send()
  })
}
