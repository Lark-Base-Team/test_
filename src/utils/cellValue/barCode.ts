

export function barCode(value: string) {
  return {
    text: String(value),
    type: "text",
  }
};