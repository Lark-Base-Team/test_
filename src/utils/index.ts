import { useDark } from "@vueuse/core"
export * from "./useBitable"

export const isDark = useDark()
