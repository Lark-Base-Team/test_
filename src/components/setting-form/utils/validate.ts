import type { fieldMap } from "@/types/types"
import { Warn } from "@/utils"

export function validateIndex(index: fieldMap[]) {
  if (index.length === 0) {
    Warn({
      title: "noIndex",
      message: "noIndex",
      notice: true,
      noticeParams: {
        text: "message.noIndex",
      },
    })
    return false
  }
  const noExcelField = index.filter((i) => !i.excel_field)
  if (noExcelField.length) {
    const names = noExcelField.map((i) => i.field.name).join(", ")
    Warn({
      title: "noExcelField",
      message: names,
      notice: true,
      noticeParams: {
        text: "message.indexNoExcelField",
        params: {
          fields: names,
        },
      },
    })
    return false
  }
  return true
}
