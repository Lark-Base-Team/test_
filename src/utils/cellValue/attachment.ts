import {
  type IAttachmentField,
  FieldType,
  bitable,
  UploadFileTaskStatus,
  IOpenAttachment,
} from "@lark-base-open/js-sdk"
import { defineTranslator, type AsyncParams, FieldRole } from "./cell"
import { downLoadFile, unique, validateUrl } from "@/utils"
import type { fieldMap } from "@/types/types"

enum DownloadStatus {
  Waiting = -1,
  Downloading = 0,
  Downloaded = 200,
  Error = 404,
}

interface FileCacheItem {
  url: string
  file: File | null
  name: string
  total: number
  loaded: number
  status: DownloadStatus
  token: string
  timeStamp: number
}

let cache: Record<string, FileCacheItem> = {}

const setCache = (item: FileCacheItem | Array<FileCacheItem>) => {
  if (Array.isArray(item)) {
    item.forEach((i) => {
      cache[i.url] = i
    })
    return
  }
  cache[item.url] = item
}

/**
 * Download files
 *
 * @param urls
 * @param options
 * @returns
 */
function processDownloadFiles(
  urls: Array<string>,
  options?: {
    onProgress?: AsyncParams<string>["onProgress"]
    onError?: AsyncParams<string>["onError"]
    parallel?: number
    requestConfig?: fieldMap["config"]["requestConfig"]
  },
) {
  const cacheItems = unique(urls)
    .map((url) => {
      if (!validateUrl(url)) return
      const item =
        cache[url] ||
        ({
          url,
          file: null,
          name: "",
          total: 0,
          loaded: 0,
          status: DownloadStatus.Waiting,
          token: "",
        } as FileCacheItem)
      return item
    })
    .filter((item) => {
      return (
        item &&
        [DownloadStatus.Waiting, DownloadStatus.Error].includes(item.status)
      )
    }) as Array<FileCacheItem>
  if (!cacheItems.length) return
  setCache(cacheItems)
  const { onProgress, parallel = 4, onError } = options ?? {}
  const total = cacheItems.length
  let loaded = 0
  let runningNum = 0
  const pendingItems: Array<FileCacheItem> = []
  const _run = () => {
    while (runningNum < parallel && cacheItems.length) {
      const item = cacheItems.shift()
      if (!item) return
      item.status = DownloadStatus.Downloading
      runningNum++
      pendingItems.push(item)
      downLoadFile(item.url, {
        onProgress: (p) => {
          console.log("processDownloadFiles", p)
          item.loaded = p.loaded
          item.total = p.total
          onProgress?.({
            total,
            loaded,
            message: pendingItems
              .map(
                (i) =>
                  `Downloading ${i.url} ${((i.loaded / i.total) * 100).toFixed(
                    2,
                  )}%`,
              )
              .join("\n"),
          })
        },
        fetchOptions: options?.requestConfig,
      })
        .then((file) => {
          item.file = file
          item.name = file?.name ?? ""
          item.status = DownloadStatus.Downloaded
          console.log("res", item, file)
        })
        .catch((e) => {
          item.status = DownloadStatus.Error
          onError?.(e)
          console.error(e)
        })
        .finally(() => {
          runningNum--
          loaded++
          console.log("cache", cache)
          pendingItems.splice(pendingItems.indexOf(item), 1)
          console.log("cache", cache)
          _run()
        })
    }
  }
  return new Promise((resolve) => {
    console.log("processDownloadFiles", cacheItems)
    _run()
    const timer = setInterval(() => {
      if (!cacheItems.length) {
        clearInterval(timer)
        resolve(null)
      }
    }, 1000)
  })
}

async function normalization(value: string, config?: fieldMap["config"]) {
  const { separator = "," } = config || {}
  const urls = value.split(separator)
  return urls
}

async function attachment(value: string, field: IAttachmentField) {
  const urls = await normalization(value)
  const files = urls
    .map((url) => {
      const item = cache[url]
      if (!item) return null
      return {
        name: item.name,
        token: item.token,
        size: item.total,
        type: item.file?.type ?? "",
        timeStamp: new Date().getTime(),
      }
    })
    .filter((item) => item) as IOpenAttachment[]
  if (!files) return null
  return await field.createCell(files)
}

export const AttachmentTranslator = defineTranslator({
  fieldType: FieldType.Attachment,
  translate: attachment,
  normalization: async (value: string) => value,
  name: "Attachment",
  asyncMethod: async (
    options: AsyncParams<string>,
    config?: fieldMap["config"],
  ) => {
    const { data: urls, onProgress, onError } = options
    const { requestConfig } = config || {}
    if (!urls.length) return
    await processDownloadFiles(urls, { onProgress, onError, requestConfig })
    const hasFileItems = Object.values(cache).filter((item) => item.file)
    const files = hasFileItems.map((item) => item.file) as File[]
    const off = bitable.base.onUploadStatusChange(({ data }) => {
      const { tasks } = data
      const { list } = tasks
      onProgress?.({
        loaded: list.filter((t) => t.status === UploadFileTaskStatus.Success)
          .length,
        total: list.length,
        message: list
          .filter((t) => t.status === UploadFileTaskStatus.Pending)
          .map(
            (t) =>
              `Uploading ${t.name} ${((t.uploadedSize / t.size) * 100).toFixed(
                2,
              )}`,
          )
          .join("\n"),
      })
    })
    const tokens = await bitable.base.batchUploadFile(files)
    let index = 0
    const items = Object.values(hasFileItems)
    for (const t in items) {
      const item = items[t]
      if (!item.file) continue
      item.token = tokens[index]
      index += 1
    }
    off()
  },
  roles: [FieldRole.ASYNC],
  refresh: () => {
    cache = {}
  },
})
