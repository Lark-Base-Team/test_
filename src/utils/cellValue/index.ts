import {
  IOpenCellValue,
  FieldType,
  IMultiSelectFieldMeta,
  ISingleSelectFieldMeta,
} from "@lark-base-open/js-sdk";
import { fieldMap } from "@/types/types";
import { dateTime, dateDefaultFormat } from "./date";
import { multiSelect } from "./multiSelect";
import { singleSelect } from "./singleSelect";
import { checkBox, defaultBoolValue } from "./checkBox";
import { text } from "./text";
import { url } from "./url";
import { phone } from "./phone";
import { currency } from "./currency";
import { progress } from "./progress";
import { rating } from "./rating";
import { barCode } from "./barCode";
import {User} from "./user";

export function getCellValue(
  fieldMap: fieldMap,
  value: string,
): IOpenCellValue {
  const type = fieldMap.field.type;
  const field = fieldMap.field;
  const config = fieldMap.config;
  if (!value) return value;
  switch (type) {
    case FieldType.Url:
      return url(value);
    case FieldType.DateTime:
      return dateTime(value, config?.format ?? dateDefaultFormat);
    case FieldType.SingleSelect:
      return singleSelect(value, field as ISingleSelectFieldMeta);
    case FieldType.MultiSelect:
      return multiSelect(
        value,
        field as IMultiSelectFieldMeta,
        config?.separator ?? ","
      );
    case FieldType.Number:
      return Number(value.match(/-?\d+\.?\d*/g));
    case FieldType.Currency:
      return currency(value);
    case FieldType.Progress:
      return progress(value);
    case FieldType.Rating:
      return rating(value);
    case FieldType.Checkbox:
      return checkBox(value, config?.boolValue ?? defaultBoolValue);
    case FieldType.Text:
      return text(value) as IOpenCellValue;
    case FieldType.Barcode:
      return barCode(value) as IOpenCellValue;
    case FieldType.Phone:
      return phone(value);
    case FieldType.User:
      return User(value);
    default:
      return value;
  }
}
