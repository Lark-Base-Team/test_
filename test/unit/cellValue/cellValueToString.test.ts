import { cellValueToString } from "../../../src/utils/cellValue"
import { describe, it, expect } from "vitest"
import {
  IOpenLink,
  IOpenLocation,
  IOpenSegment,
  IOpenSegmentType,
  IOpenSingleSelect,
  IOpenUser,
} from "@lark-base-open/js-sdk"
const segment: IOpenSegment[] = [
  {
    type: IOpenSegmentType.Text, // 文本类型
    text: "Hello world",
  }, // type: IOpenTextSegment
  {
    type: IOpenSegmentType.Url, // URL 类型
    text: "https://feishu.cn",
    link: "https://feishu.cn",
  }, // type: IOpenUrlSegment
  {
    type: IOpenSegmentType.Mention, // @人 类型
    mentionType: "User", //
    name: "李健", // 人名
    enName: "Jian Li",
    text: "@李健",
    token: "ou_xxx",
    id: "ou_xxx", // OpenId，暂未支持
  }, // type: IOpenUserMentionSegment
  {
    type: IOpenSegmentType.Mention, // 文档类型
    mentionType: "Bitable", // Bitable 文档
    text: "未命名多维表格",
    token: "basxxxx",
    link: "https://www.feishu.cn/base/basxxxxx",
  }, // type: IOpenDocumentMentionSegment
]
const singleSelect: IOpenSingleSelect = {
  id: "opt123", //这里的id不是随便填的，是从字段meta信息的property属性中选的一个值
  text: "选项1",
}

const multiSelect: IOpenSingleSelect[] = [
  {
    id: "opt123",
    text: "选项1",
  },
  {
    id: "opt456",
    text: "选项2",
  },
]

const user: IOpenUser[] = [
  {
    name: "@李健", // 人名
    en_name: "@JianLi", // 英文名
    email: "lijian@feishu.cn", // 邮箱
    id: "ou_xxx", // OpenId
  },
]

const location: IOpenLocation = {
  address: "科苑地铁站C出口",
  adname: "南山区",
  cityname: "深圳市",
  fullAddress: "深圳湾创新科技中心，广东深圳南山科苑地铁站C出口",
  full_address: "深圳湾创新科技中心，广东深圳南山科苑地铁站C出口",
  location: "113.946874,22.52835",
  name: "深圳湾创新科技中心",
  pname: "广东省",
}

const link: IOpenLink[] = [
  {
    record_ids: ["recMQTzZP6"],
    recordIds: ["recMQTzZP6"],
    tableId: "tblHBG6ACPXeTu2G",
    table_id: "tblHBG6ACPXeTu2G",
    text: "1",
    type: "text",
  },
]

describe("cellValueToString test", () => {
  it("IOpenSegment->string", () => {
    expect(cellValueToString(segment)).toMatchSnapshot()
  })

  it("TimeStamp->string", () => {
    expect(cellValueToString(1614048000000)).toMatchSnapshot()
  })

  it("number->string", () => {
    expect(cellValueToString(1)).toMatchSnapshot()
  })

  it("checkbox->string", () => {
    expect(cellValueToString(true)).toMatchSnapshot()
  })

  it("IOpenSingleSelect->string", () => {
    expect(cellValueToString(singleSelect)).toMatchSnapshot()
  })

  it("IOpenMultiSelect->string", () => {
    expect(cellValueToString(multiSelect)).toMatchSnapshot()
  })

  it("IOpenUser->string", () => {
    expect(cellValueToString(user)).toMatchSnapshot()
  })

  it("IOpenLocation->string", () => {
    expect(cellValueToString(location)).toMatchSnapshot()
  })

  it("IOpenLink->string", () => {
    expect(cellValueToString(link)).toMatchSnapshot()
  })
})
