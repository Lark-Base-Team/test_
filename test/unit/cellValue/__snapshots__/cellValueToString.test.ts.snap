// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cellValueToString test > IOpenLink->string 1`] = `"1"`;

exports[`cellValueToString test > IOpenLocation->string 1`] = `"深圳湾创新科技中心，广东深圳南山科苑地铁站C出口"`;

exports[`cellValueToString test > IOpenMultiSelect->string 1`] = `"选项1,选项2"`;

exports[`cellValueToString test > IOpenSegment->string 1`] = `"Hello worldhttps://feishu.cn@李健未命名多维表格"`;

exports[`cellValueToString test > IOpenSingleSelect->string 1`] = `"选项1"`;

exports[`cellValueToString test > IOpenUser->string 1`] = `"@李健"`;

exports[`cellValueToString test > TimeStamp->string 1`] = `"1614048000000"`;

exports[`cellValueToString test > checkbox->string 1`] = `"☑️"`;

exports[`cellValueToString test > number->string 1`] = `"1"`;
